### PR TITLE
Add Timer.Reset()

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by benbjohnson** [comment]
Can you provide some details or a use case for why this functionality should be added?

**Comment by nomis52** [comment]
It's in go library: https://golang.org/pkg/time/#Timer.Reset

As for justification: we can't use this clock library for testing if the code calls timer.Reset()

File: clock.go [hunk 1]
```diff
@@ -231,21 +232,42 @@ func (a clockTimers) Less(i, j int) bool { return a[i].Next().Before(a[j].Next()
...
-	next  time.Time   // next tick time
-	mock  *Mock       // mock clock, if set
-	fn    func()      // AfterFunc function, if set
+	C          <-chan time.Time
+	c          chan time.Time
+	timer      *time.Timer // realtime impl, if set
+	next       time.Time   // next tick time
+	mock       *Mock       // mock clock, if set
+	fn         func()      // AfterFunc function, if set
+	registered bool        // True if active, false if stopped
```
Comment (benbjohnson) [inline clock.go]: Can you rename this to `stopped` and flip the `true`/`false` values? e.g. `// True if stopped, false if active`.

File: clock.go [hunk 2]
```diff
@@ -231,21 +232,42 @@ func (a clockTimers) Less(i, j int) bool { return a[i].Next().Before(a[j].Next()
...
-func (t *Timer) Stop() {
+func (t *Timer) Stop() bool {
 	if t.timer != nil {
-		t.timer.Stop()
-	} else {
-		t.mock.removeClockTimer((*internalTimer)(t))
+		return t.timer.Stop()
+	}
+
+	wasRegistered := t.registered
```
Comment (benbjohnson) [inline clock.go]: Can you rename `wasRegistered` to `registered`?

File: clock.go [hunk 3]
```diff
@@ -231,21 +232,42 @@ func (a clockTimers) Less(i, j int) bool { return a[i].Next().Before(a[j].Next()
...
+}
+
+// Reset changes the expiry time of the timer
+func (t *Timer) Reset(d time.Duration) bool {
+	if t.timer != nil {
+		return t.timer.Reset(d)
+	}
+
+	t.next = t.mock.now.Add(d)
+	wasRegistered := t.registered
```
Comment (benbjohnson) [inline clock.go]: Rename `wasRegistered` to `registered`.

**Comment by benbjohnson** [comment]
@nomis52 Cool, makes sense. Minor naming feedback but otherwise lgtm.

File: clock.go (hunk 3 above)
Comment (nomis52) [inline clock.go]: Done

File: clock.go (hunk 2 above)
Comment (nomis52) [inline clock.go]: Done

File: clock.go (hunk 1 above)
Comment (nomis52) [inline clock.go]: Done

**Pull request merged**